src/domains/users/users.handlers.ts

```diff
@@ -0,0 +1,103 @@
+import type { z } from '@hono/zod-openapi';
+import { eq, or } from 'drizzle-orm';
+
+import type { insertUsersSchema, patchUsersSchema, selectUsersSchema } from '@/db/schema';
+import { users } from '@/db/schema';
+import { db } from '@/db';
+
+export type User = z.infer<typeof selectUsersSchema>;
+export type CreateUserInput = z.infer<typeof insertUsersSchema>;
+export type UpdateUserInput = z.infer<typeof patchUsersSchema>;
+
+export type Result<T, E = { message: string }> = 
+  | { ok: true; data: T }
+  | { ok: false; error: E };
+
+export async function getAllUsers(): Promise<User[]> {
+  return await db.select().from(users);
+}
+
+export async function getUserById(id: string): Promise<User | null> {
+  const result = await db.select().from(users).where(eq(users.id, id)).limit(1);
+  return result[0] || null;
```

Comment:
Since we are using the `Result` return type we should try to keep the api consistent for this service (handler) in all available functions.
```suggestion
  return user 
    ? { ok: true, data: user }
    : { ok: false, error: { message: 'User not found' } };
```